### PR TITLE
KAFKA-16331: Remove EOSv1 from Kafka Streams system tests

### PR DESCRIPTION
**Pull request by mjsax**
We will remove EOSv1 in 4.0, and thus can remove the corresponding system tests.

### CODE REVIEW

**Comment by mjsax** [comment]
Triggered a system test run to verify that we did not break anything: https://semaphore.ci.confluent.io/workflows/4481964d-cbd3-4982-959f-98d9f2cfa4bf

File: streams/src/test/java/org/apache/kafka/streams/tests/BrokerCompatibilityTest.java [hunk 1]
```diff
@@ -68,8 +68,8 @@ public static void main(final String[] args) throws IOException {
 
         streamsProperties.put(StreamsConfig.APPLICATION_ID_CONFIG, "kafka-streams-system-test-broker-compatibility");
         streamsProperties.put(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "earliest");
-        streamsProperties.put(StreamsConfig.DEFAULT_KEY_SERDE_CLASS_CONFIG, Serdes.String().getClass());
-        streamsProperties.put(StreamsConfig.DEFAULT_VALUE_SERDE_CLASS_CONFIG, Serdes.String().getClass());
+        streamsProperties.put(StreamsConfig.DEFAULT_KEY_SERDE_CLASS_CONFIG, Serdes.StringSerde.class);
+        streamsProperties.put(StreamsConfig.DEFAULT_VALUE_SERDE_CLASS_CONFIG, Serdes.StringSerde.class);
```
Comment (mjsax) [inline streams/src/test/java/org/apache/kafka/streams/tests/BrokerCompatibilityTest.java:72]: Side improvement

**Comment by mjsax** [comment]
`streams_broker_compatibility_test` and `streams_eos_test` failed. Pushed a fix and triggered re-runs
 - https://semaphore.ci.confluent.io/workflows/87fd8199-6507-4e40-83b7-2e9500f87626
 - https://semaphore.ci.confluent.io/workflows/b08bdfcd-6c7f-4ae1-8e92-dead9c121fcd

**Comment by mjsax** [comment]
Another re-run: https://semaphore.ci.confluent.io/workflows/82a881c7-93f6-4a1e-91a5-d0fb634b3212?pipeline_id=af4f87e6-1aa3-4300-9993-6f23774e0169

File: streams/src/test/java/org/apache/kafka/streams/tests/StreamsEosTest.java [hunk 2]
```diff
@@ -49,12 +48,9 @@ public static void main(final String[] args) throws IOException {
...
 
         if ("process".equals(command) || "process-complex".equals(command)) {
-            if (!StreamsConfig.EXACTLY_ONCE.equals(processingGuarantee) &&
-                !StreamsConfig.EXACTLY_ONCE_BETA.equals(processingGuarantee) &&
-                !StreamsConfig.EXACTLY_ONCE_V2.equals(processingGuarantee)) {
+            if (!StreamsConfig.EXACTLY_ONCE_V2.equals(processingGuarantee)) {
 
-                System.err.println("processingGuarantee must be either " + StreamsConfig.EXACTLY_ONCE + " or " +
-                                       StreamsConfig.EXACTLY_ONCE_BETA + " or " + StreamsConfig.EXACTLY_ONCE_V2);
+                System.err.println("processingGuarantee must be " + StreamsConfig.EXACTLY_ONCE_V2);
```
Comment (chia7712) [inline streams/src/test/java/org/apache/kafka/streams/tests/StreamsEosTest.java:53]: nit: `processingGuarantee` -> `processing.guarantee`

File: streams/src/test/java/org/apache/kafka/streams/tests/StreamsSmokeTest.java [hunk 3]
```diff
@@ -60,14 +59,10 @@ public static void main(final String[] args) throws IOException {
 
         if ("process".equals(command)) {
             if (!StreamsConfig.AT_LEAST_ONCE.equals(processingGuarantee) &&
-                !StreamsConfig.EXACTLY_ONCE.equals(processingGuarantee) &&
-                !StreamsConfig.EXACTLY_ONCE_BETA.equals(processingGuarantee) &&
                 !StreamsConfig.EXACTLY_ONCE_V2.equals(processingGuarantee)) {
 
                 System.err.println("processingGuarantee must be either " +
```
Comment (chia7712) [inline streams/src/test/java/org/apache/kafka/streams/tests/StreamsSmokeTest.java:64]: nit: `processingGuarantee` -> `processing.guarantee`

**Review by chia7712 — COMMENTED** [review]
@mjsax thanks for this cleanup. some trivial comments are left.

File: streams/src/test/java/org/apache/kafka/streams/tests/StreamsEosTest.java (hunk 2 above)
Comment (mjsax) [reply streams/src/test/java/org/apache/kafka/streams/tests/StreamsEosTest.java:53]: Actually correct as-is, as we refer to the variable `processingGuarantee` here -- note: it's internal test code; not user facing.

File: streams/src/test/java/org/apache/kafka/streams/tests/StreamsSmokeTest.java (hunk 3 above)
Comment (mjsax) [reply streams/src/test/java/org/apache/kafka/streams/tests/StreamsSmokeTest.java:64]: Same.

File: streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java [hunk 4]
```diff
@@ -742,9 +742,9 @@ boolean runLoop() {
                     errorMessage.startsWith("Broker unexpectedly doesn't support requireStable flag on version ")) {
 
                     log.error("Shutting down because the Kafka cluster seems to be on a too old version. " +
-                              "Setting {}=\"{}\"/\"{}\" requires broker version 2.5 or higher.",
+                              "Setting {}=\"{}\" requires broker version 2.5 or higher.",
                           StreamsConfig.PROCESSING_GUARANTEE_CONFIG,
-                          StreamsConfig.EXACTLY_ONCE_V2, StreamsConfig.EXACTLY_ONCE_BETA);
```
Comment (chia7712) [inline streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java:747]: Could you please remove the `@SuppressWarnings("deprecation")` from line:688 as `StreamsConfig.EXACTLY_ONCE_BETA` is removed

**Review by chia7712 — APPROVED** [review]
LGTM except for one small comment.

**Review by bbejeck — APPROVED** [review]
LGTM

File: streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java (hunk 4 above)
Comment (mjsax) [reply streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java:747]: Good point. Will do in a follow up PR. (EOSv1 removal will be many PRs...)

Comment (mjsax) [reply streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java:747]: Addressed via https://github.com/apache/kafka/pull/17170

**Pull request merged**